bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/EnergyManagerHandler.java

```diff
@@ -0,0 +1,402 @@
+        if (timestamp.compareTo(BigDecimal.ONE) > 0) {
+            energyManager.addState(CHANNEL_DATETIME.getChannelName(),
+                    new DateTimeType(this.getFromMilliTimestamp(timestamp)));
+        }
+    }
+
+    /**
+     * Initial setup of the channels available for this thing.
+     *
+     * @param device which provides the channels
+     */
+    protected void initDeviceChannels(Device device) {
+        this.logger.debug("{}: initDeviceChannels for device {}", this, this.getThing().getUID());
+
+        this.assertChannel(new SolarwattChannel(CHANNEL_DATETIME.getChannelName(), "time"));
+
+        device.getSolarwattChannelSet().forEach((channelTag, solarwattChannel) -> {
+            this.logger.trace("{}: {}", this.getThing().getUID(), solarwattChannel.getChannelName());
+            this.assertChannel(solarwattChannel);
+        });
+    }
+
+    /**
+     * Assert that all channels inside of our thing are well defined.
+     *
+     * Only channel which can not be found are created.
+     *
+     * @param solarwattChannel channel description with name and unit
+     */
+    protected void assertChannel(SolarwattChannel solarwattChannel) {
+        ChannelUID channelUID = new ChannelUID(this.getThing().getUID(), solarwattChannel.getChannelName());
+        ChannelTypeUID channelType = this.channelTypeProvider.assertChannelType(solarwattChannel);
+        if (this.getThing().getChannel(channelUID) == null) {
+            ThingBuilder thingBuilder = this.editThing();
+            thingBuilder.withChannel(
+                    SimpleDeviceHandler.getChannelBuilder(solarwattChannel, channelUID, channelType).build());
+
+            this.updateThing(thingBuilder.build());
+        }
+    }
+
+    /**
+     * Finds the guid of the energy manager inside of the known devices.
+     *
+     * @param devices list with known devices
+     * @throws SolarwattConnectionException if there is no energy manager available
+     */
+    private void findEnergyManagerGuid(Map<String, Device> devices) throws SolarwattConnectionException {
+        devices.forEach((guid, device) -> {
+            if (device instanceof EnergyManager) {
+                this.energyManagerGuid = guid;
+            }
+        });
+
+        if (this.energyManagerGuid == null) {
+            throw new SolarwattConnectionException("unable to find energy manager");
+        }
+    }
+
+    /**
+     * Setup the handler and trigger initial load via {@link EnergyManagerHandler::refreshDevices}.
+     *
+     * Web request against energy manager and loading of devices is deferred and will send the ONLINE
+     * event after loading all devices.
+     */
+    @Override
+    public void initialize() {
+        this.logger.debug("{} initialize", this);
+        SolarwattBridgeConfiguration localConfig = this.getConfigAs(SolarwattBridgeConfiguration.class);
+        this.initRefresh(localConfig);
+        this.initDeviceCache(localConfig);
```

Comment:
If I see correctly this can lead to network operation, but this method must return fast. You could schedule a task for this operation.

Reply:
initRefresh is using a ScheduledFuture to perform the network operation and the initDevicesCache is using the scheduler to start the first network operation.

Reply:
MAybe I use the wrong maven target, but  target/code-analysis/report.html is not generated on my local machine.

Reply:
If you call `mvn clean install` it will be generated here `openhab-addons/bundles/org.openhab.binding.solarwatt/target/code-analysis/report.html`.

Reply:
With just using `mvn clean package` the code-analysis was not started. Now the findings are fixed.